app.js

```diff
@@ -1,8 +1,166 @@
+                    alert(gobbie.name + ' got hit');
+                    //else argument if the 50% chance fails
+                } else {
+                  //alert populates notifying that the player missed the clicked gobbie
+                    alert('Uplander missed ' + gobbie.name);
+                }
+                //after player attack executes regardless of outcome, gobbie attacks with a 33% chance from math.random
+                if (Math.random() > gobbie.accuracy) {
+                  //results of if statement, player loses hitpoints equal to gobbies randomly generated strength
+                    playerHP -= gobbie.strength;
+                    //alert populates notifying player that the gobbie successfully hit them
+                    alert(gobbie.name + ' hit the Uplander');
+                } else {
+                  //if gobbie fails 33% to hit, alert populates with string concatenated with gobbie's inputed name notifying that the gobbie successfully hit
+                    alert('Uplander dodged ' + gobbie.name + '\'s attack!');
+                }
+                //if statement based on if gobbie's hitpoints reaches zero
+                if (gobbie.hp <= 0) {
+                  //counter for number of gobbies defeated increments by one
+                    gobbiefolkDefeated++;
+                    playerExp++;
+                    levelUp();
+                    //h3 element set to update with concatenated string with of count of defeated gobbies
+                    defeatedGobbiesEl.textContent = `Uplander's murderkilled ${gobbiefolkDefeated} innocent gobbiefolk(s)`;
+                }
+                if (gobbiefolkDefeated === 3 && hpPotion === 0 || gobbiefolkDefeated === 7 && hpPotion === 0 || gobbiefolkDefeated === 12 && hpPotion === 0 || gobbiefolkDefeated === 17 && hpPotion === 0)
+                {
+                    hpPotion++;
+                }
+                potionEl.textContent = `Healing Potions: ${hpPotion}`;
+            }
+                //if statement declaring results if player's health reaches or goes below zero
+            if (playerHP <= 0) {
+              //populates an alert stating that the game is over (currently no effect on "gameplay")
+                alert('You and your friends are dead. Game over');
+            }
+            //calls the displayGobbies function within the eventListener inside the displayGobbies function to execute
+            displayGobbies();
+            displayPlayer();
+            //updates HTML element with player health if necessary
+            playerHPEl.textContent = `Uplander HP is currently: ${playerHP}`;
+            //updates HTML element with player exp if necessary
+            playerExpEl.textContent = `Uplander EXP: ${playerExp}`;
+            
+            
+        });
+        //adds the dynamic gobbie div created in this function to the gobbie list div
+
```

Comment:
I absolutely LOVE the pseudo code above each line of code. So good to see you going through what you want to do in english and then the corresponding code below to execute the vision. Great work.